### PR TITLE
[NFC] [SYCL] Make code a bit cleaner when using DISABLE_SYCL_INSTRUMENTATION_METADATA

### PR DESCRIPTION
**Pull request by s-kanaev**

### CODE REVIEW

File: sycl/include/CL/sycl/queue.hpp [hunk 1]
```diff
@@ -22,6 +22,39 @@
...
+// detail::code_location::current()
+// replace _CODELOCARG(&CodeLoc) with nothing
+// or :  const detail::code_location &CodeLoc = {}
+
+#ifndef DISABLE_SYCL_INSTRUMENTATION_METADATA
+#define _CODELOCPARAM(a)                                                       \
+  , const detail::code_location a = detail::code_location::current()
+
+#define _CODELOCARG(a)
+#define _CODELOCFW(a) , a
```
Comment (bader) [inline sycl/include/CL/sycl/queue.hpp:42]: This must be undefined as well.
See https://github.com/intel/llvm/blob/a42d4a4d99bb800f25a0d70e8c84be4fa991584f/sycl/include/CL/sycl/queue.hpp#L711-L714

**Review by bader — CHANGES_REQUESTED** [review]

File: sycl/include/CL/sycl/queue.hpp (hunk 1 above)
Comment (s-kanaev) [reply sycl/include/CL/sycl/queue.hpp:42]: Oh, good catch.
Done.

**Comment by s-kanaev** [comment]
> Please, apply clang-format changes to fix CI check.

Done.

File: sycl/include/CL/sycl/queue.hpp [hunk 2]
```diff
@@ -22,6 +22,41 @@
...
+// these macros are #undef immediately.
+
+// replace _CODELOCPARAM(&CodeLoc) with nothing
+// or :   , const detail::code_location &CodeLoc =
+// detail::code_location::current()
+// replace _CODELOCARG(&CodeLoc) with nothing
+// or :  const detail::code_location &CodeLoc = {}
+
+#ifndef DISABLE_SYCL_INSTRUMENTATION_METADATA
+#define _CODELOCONLYPARAM(a)                                                   \
```
Comment (romanovvlad) [inline sycl/include/CL/sycl/queue.hpp:38]: It seems the `_CODELOCONLYPARAM` is defined only if `DISABLE_SYCL_INSTRUMENTATION_METADATA` is not defined while used unconditionally on line 275.

**Review by tovinkere — DISMISSED** [review]
Suggestion: If we are going to turn it off by default and manage enabling through compiler command-line define to ensure it is an opt-in capability, it would make sense to reverse the logic and the macro name to ENABLE_SYCL_INSTRUMENTATION_METADATA.  If this can be hidden by the compiler front end, then I suppose it shouldn't matter.

LGTM otherwise.

File: sycl/include/CL/sycl/queue.hpp (hunk 2 above)
Comment (s-kanaev) [reply sycl/include/CL/sycl/queue.hpp:38]: Fixed.

**Comment by s-kanaev** [comment]
I think it should be done as distinct patch

**Review by romanovvlad — APPROVED** [review]

**Pull request merged**